staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -225,6 +226,74 @@ func convertPodNamedPortToNumber(ports []string, pod corev1.Pod) ([]string, erro
 	return converted, nil
 }
 
+// checkUDPPortInService port returns error if remote port in Service is a UDP port
+// TODO: remove this check after #47862 is resolved
+func checkUDPPortInService(ports []string, svc corev1.Service) error {
+	udpOnlyPorts := sets.NewInt()
+	for _, port := range svc.Spec.Ports {
+		portNum := int(port.Port)
+		switch port.Protocol {
+		case corev1.ProtocolUDP:
+			udpOnlyPorts.Insert(portNum)
+		case corev1.ProtocolTCP:
+			if udpOnlyPorts.Has(portNum) {
+				udpOnlyPorts.Delete(portNum)
+			}
+		default:
+		}
+	}
+	for _, port := range ports {
+		_, remotePort := splitPort(port)
+		portNum, err := strconv.Atoi(remotePort)
+		if err != nil {
+			svcPort, err := util.LookupServicePortNumberByName(svc, remotePort)
+			if err != nil {
+				return err
+			}
+			portNum = int(svcPort)
+		}
+		if udpOnlyPorts.Has(portNum) {
+			return fmt.Errorf("UDP protocol is not supported now: %s", remotePort)
+		}
+	}
+	return nil
+}
+
+// checkUDPPortInPod port returns error if remote port in Pod is a UDP port
```

Comment:
For me the word `port` after checkUDPPortInPod sounds strange to me. I'm not sure, but I think this makes more sense:
```diff
- // checkUDPPortInPod port returns error if remote port in Pod is a UDP port
+ // checkUDPPortInPod returns an error if remote port in Pod is an UDP port
```

Reply:
oh it is a typo i think.

Reply:
I also suggested two other things:
```diff
- a error
+ an error

- a UDP port
- an UDP port
```

I'm not an english native, but I think it's correct :)

Reply:
neither an englist native. it should be "an error", but i think "a udp port" is correct.